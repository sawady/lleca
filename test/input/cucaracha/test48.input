
fun f(b : Bool) {
    if b {
        x := 100
        y := 10
    } else {
        y := 99
        x := 999
    }
    putNum(x - y)
    putChar(10)
}

fun g() {
    if True {
        x := 10
    }
    putNum(x)
    putChar(10)
}

fun main() {
    f(True)
    f(False)
    g()
}

