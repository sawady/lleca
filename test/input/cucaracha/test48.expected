Cons(
  Function(
    f,
    Cons(
      Parameter(
        b,
        Bool
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtIf(
        ExprVar(
          b
        ),
        Cons(
          StmtAssign(
            x,
            ExprConstNum(
              100
            )
          ),
          Cons(
            StmtAssign(
              y,
              ExprConstNum(
                10
              )
            ),
            Nil
          )
        ),
        Just(
          Cons(
            StmtAssign(
              y,
              ExprConstNum(
                99
              )
            ),
            Cons(
              StmtAssign(
                x,
                ExprConstNum(
                  999
                )
              ),
              Nil
            )
          )
        )
      ),
      Cons(
        StmtCall(
          putNum,
          Cons(
            ExprSub(
              ExprVar(
                x
              ),
              ExprVar(
                y
              )
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Nil
        )
      )
    )
  ),
  Cons(
    Function(
      g,
      Nil,
      Unit,
      Cons(
        StmtIf(
          ExprConstBool(
            True
          ),
          Cons(
            StmtAssign(
              x,
              ExprConstNum(
                10
              )
            ),
            Nil
          ),
          Nothing
        ),
        Cons(
          StmtCall(
            putNum,
            Cons(
              ExprVar(
                x
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  10
                ),
                Nil
              )
            ),
            Nil
          )
        )
      )
    ),
    Cons(
      Function(
        main,
        Nil,
        Unit,
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstBool(
                True
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              f,
              Cons(
                ExprConstBool(
                  False
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                g,
                Nil
              ),
              Nil
            )
          )
        )
      ),
      Nil
    )
  )
)
